sale_commission/wizard/wizard_settle.py

```diff
@@ -85,11 +85,15 @@ def action_settle(self):
                         settlement = settlement_obj.create(
                             {'agent': agent.id,
                              'date_from': sett_from,
-                             'date_to': sett_to})
+                             'date_to': sett_to,
+                             'company_id':
+                                agent_lines[pos].invoice_line.company_id.id})
```

Comment:
I think that is not necessary, it is the same problem that invoice by picking, the user must stay in the company that try to invoice.